server/src/main/scala/drt/server/feeds/edi/EdiFeed.scala

```diff
@@ -0,0 +1,154 @@
+package drt.server.feeds.edi
+
+import akka.NotUsed
+import akka.actor.{ActorSystem, Cancellable}
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
+import akka.http.scaladsl.model.HttpResponse
+import akka.http.scaladsl.model.StatusCodes.OK
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.stream.Materializer
+import akka.stream.scaladsl.Source
+import drt.shared.FlightsApi.Flights
+import drt.shared.Terminals.Terminal
+import drt.shared.api.{Arrival, FlightCodeSuffix}
+import drt.shared.{FeedSource, _}
+import org.joda.time.{DateTime, DateTimeZone}
+import org.slf4j.{Logger, LoggerFactory}
+import server.feeds.{ArrivalsFeedFailure, ArrivalsFeedResponse, ArrivalsFeedSuccess}
+import services.SDate
+import services.SDate.JodaSDate
+import spray.json.{DefaultJsonProtocol, RootJsonFormat}
+
+import scala.collection.immutable.Seq
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+
+trait EdiFeedJsonSupport extends DefaultJsonProtocol with SprayJsonSupport {
+
+  implicit val ediFlightDetailsFormat: RootJsonFormat[EdiFlightDetails] = jsonFormat22(EdiFlightDetails)
+
+  def unMarshalResponseToEdiFlightDetails(httpResponse: HttpResponse)(implicit system: ActorSystem, mat: Materializer, ec: ExecutionContext): Future[List[EdiFlightDetails]] = {
+    httpResponse.entity.dataBytes.runReduce(_ ++ _)
+      .flatMap(d => Unmarshal(d.utf8String).to[List[EdiFlightDetails]])
+  }
+}
+
+class EdiFeed(ediClient: EdiClient) extends EdiFeedJsonSupport {
+
+  val log: Logger = LoggerFactory.getLogger(getClass)
+
+  def makeRequestAndFeedResponseToArrivalSource(startData: String, endDate: String, feedSource: FeedSource)(implicit system: ActorSystem, mat: Materializer, ec: ExecutionContext): Future[ArrivalsFeedResponse] = {
+    log.info(s"$feedSource Edi feed api call at ${DateTime.now()} from $startData and $endDate ")
+    ediClient.makeRequest(startData, endDate).flatMap { response =>
+      response.status match {
+        case OK => unMarshalResponseToEdiFlightDetails(response).map { flights =>
+          log.info(s"Edi feed status ${response.status} with api call for ${flights.size} flights")
+          ArrivalsFeedSuccess(Flights(ediFlightDetailsToArrival(flights, feedSource)))
+        }
+        case _ => log.warn(s"Edi feed status ${response.status} while api call with response ${response.entity}")
+          Future.successful(ArrivalsFeedFailure(s"Response with status ${response.status} from edi ${response.entity}"))
+      }
+    }.recover { case t =>
+      log.error(s"Edi feed error while api call ${t.getMessage}")
+      ArrivalsFeedFailure(t.getMessage)
+    }
+  }
+
+  def ediFeedPollingSource(interval: FiniteDuration, feedSource: FeedSource)(implicit system: ActorSystem, mat: Materializer, ec: ExecutionContext): Source[ArrivalsFeedResponse, Cancellable] = {
+    var startForecastDays = 2
```

Comment:
Why do we have `var`s here?

Reply:
there is restriction of 30 days in one api calls , so for 180 days we need date get data for 30 days incrementally . 

May be I can try without var but in first iteration I find var to simple to implement .